viam-orb-slam3.go

```diff
@@ -269,64 +261,50 @@ func (slamSvc *builtIn) Position(ctx context.Context, name string, extra map[str
 
 	var pInFrame *referenceframe.PoseInFrame
 	var returnedExt map[string]interface{}
+	var componentReference string
 
 	// TODO: Once RSDK-1053 (https://viam.atlassian.net/browse/RSDK-1066) is complete the original code before extracting position
 	// from GetPosition will be removed and the GetPositionNew -> GetPosition
-	if slamSvc.dev {
-		slamSvc.logger.Debug("IN DEV MODE (position request)")
+	if orbSvc.dev {
+		orbSvc.logger.Debug("IN DEV MODE (position request)")
 		req := &pb.GetPositionNewRequest{Name: name}
 
-		resp, err := slamSvc.clientAlgo.GetPositionNew(ctx, req)
+		resp, err := orbSvc.clientAlgo.GetPositionNew(ctx, req)
 		if err != nil {
 			return nil, errors.Wrap(err, "error getting SLAM position")
 		}
 
-		pInFrame = referenceframe.NewPoseInFrame(resp.GetComponentReference(), spatialmath.NewPoseFromProtobuf(resp.GetPose()))
+		componentReference = resp.GetComponentReference()
+		pInFrame = referenceframe.NewPoseInFrame(componentReference, spatialmath.NewPoseFromProtobuf(resp.GetPose()))
 		returnedExt = resp.Extra.AsMap()
 	} else {
 		req := &pb.GetPositionRequest{Name: name, Extra: ext}
 
-		resp, err := slamSvc.clientAlgo.GetPosition(ctx, req)
+		resp, err := orbSvc.clientAlgo.GetPosition(ctx, req)
 		if err != nil {
 			return nil, errors.Wrap(err, "error getting SLAM position")
 		}
 
+		componentReference = resp.Pose.ReferenceFrame
```

Comment:
I'm not sure if this is the correct way to do this. Not sure if it matters much though because this should be removed once the dev flag is removed again

Reply:
We were wondering about the same. It either returns it, or returns an empty string instead of it.

Reply:
According to Nick, this was done to better match the existing return type for `GetPosition(...)`. The function was originally just intended to encapsulate the complicated logic for quaternion math, so it was okay to do weird things like passing `componentReference` through the function unchanged.